File: hardening/sha256_oracle.php
<?php
// Independent-runtime oracle: hashes one hex-encoded artifact with PHP's own SHA-256 (ext/hash, not
// OpenSSL) and prints the digest as this runtime writes it.
// Usage: php sha256_oracle.php '<artifact bytes as hex>'
echo hash('sha256', hex2bin($argv[1])), "\n";

File: hardening/sha256_oracle.rb
# Independent-runtime oracle: hashes one hex-encoded artifact with Ruby's own SHA-256
# (ext/digest/sha2, not OpenSSL) and prints the digest as this runtime writes it.
# Usage: ruby sha256_oracle.rb '<artifact bytes as hex>'
require 'digest'
puts Digest::SHA256.hexdigest([ARGV[0]].pack('H*'))
